pages/search.js

```diff
@@ -0,0 +1,174 @@
+        </form>
+        <div className="searchButton">
+          <Button
+            variant="outlined"
+            size="medium"
+            color="primary"
+            onClick={showDataValue}
+          >
+            {' '}
+            <span role="img" aria-label="emoji">
+              🔎
+            </span>
+          </Button>
+        </div>
+      </div>
+      <div className="table">
+        <h3>
+          {/* get the sum from the data result which is as filtered length here */}
+          <span className="span">{input !== '' ? filtered.length : '0'}</span>{' '}
+          Breeds for you{' '}
+          <span role="img" aria-label="emoji">
+            💝
+          </span>
+        </h3>
+        {input === '' ? (
+          info
+        ) : (
+          // filter from the data result and create each of them as a link
+          <ul>
+            {filtered.map((name, i) => {
+              // use REGEX to match the exact id from each dog and use it as the path '/search/[id]' which below on line 91
+              const eachId = name.match(/\d/g).join('');
+              //console.log('eachId: ', eachId);
+              //console.log('name: ', name);
+
+              return (
+                <li key={i}>
+                  {/* Use ${eachId} dynamically attach each dog which has the same id, which matches each id from each name use Regex */}
+                  <Link href="/search/[id]" as={`/search/${eachId}`}>
+                    <a>{name}</a>
+                  </Link>
+                </li>
+              );
+            })}
+          </ul>
+        )}
+      </div>
+      <Footer />
+
+      <style jsx>{`
+        .searchBar {
+          display: flex;
+          justify-content: center;
+          align-items: center;
+          margin: 2em auto;
+        }
+        input {
+          width: 10em;
+          height: 2em;
+          background-color: rgb(217, 236, 230);
+          border: none;
+          border-radius: 5px;
+        }
+        .searchButton {
+          margin-left: 2em;
+        }
+        .table,
+        ul,
+        h3 {
+          text-align: center;
+          list-style: none;
+          font-family: 'Lucida Sans Unicode', 'Lucida Grande', sans-serif;
+          line-height: 2em;
+          margin: 1em auto;
+          background-color: #f8e5e8;
+          border-radius: 15px;
+          width: 60%;
+          padding: 1em;
+        }
+        .span {
+          color: #4b8ada;
+          font-size: 1.2em;
+        }
+        a {
+          text-decoration: none;
+        }
+        a:hover {
+          color: rgb(35, 174, 237);
+        }
+        @media (max-width: 450px) {
+          .table {
+            font-size: 0.7em;
+            line-height: 1em;
+            margin: 1em auto;
+            padding: 0.5;
+          }
+        }
+      `}</style>
+    </div>
+  );
+}
+
+export async function getServerSideProps(context) {
+  const { getFetchedDogsByName } = await import('../db.js');
```

Comment:
I would suggest not getting all dogs here.

Reply:
Instead of doing that, what you could do is:

1. Create a Search API route (eg. `/pages/api/search.ts`) that accepts a term via the query string (eg. `http://localhost:3000/api/search?q=bulldog` -> this shows up under `req.query.q` in your handler)
2. Use that API route when the button is clicked